SignalServiceKit/src/Messages/OWSReadReceiptManager.m

```diff
@@ -559,20 +560,45 @@ - (BOOL)areReadReceiptsEnabled
     // We don't need to worry about races around this cached value.
     if (!self.areReadReceiptsEnabledCached) {
         // Default to NO.
-        self.areReadReceiptsEnabledCached = @([self.dbConnection boolForKey:OWSReadReceiptManagerAreReadReceiptsEnabled
-                                                               inCollection:OWSReadReceiptManagerCollection]);
+        self.areReadReceiptsEnabledCached =
+            @([self.dbConnection boolForKey:OWSReadReceiptManagerAreReadReceiptsEnabled
+                               inCollection:OWSReadReceiptManagerAreReadReceiptsEnabled]);
+    }
+
+    return [self.areReadReceiptsEnabledCached boolValue];
+}
+
+- (BOOL)areReadReceiptsEnabledWithTransaction:(YapDatabaseReadTransaction *)transaction
+{
+    if (!self.areReadReceiptsEnabledCached) {
```

Comment:
Maybe add a comment about how we don't need to synchronize access to this property since, this state can only be edited locally by the user, edits will be very rare, and races don't have serious side effects.